include/aspect/particle/generator/interface.h

```diff
@@ -121,6 +138,15 @@ namespace aspect
           std::pair<types::LevelInd,Particle<dim> >
           generate_particle(const Point<dim> &position,
                             const types::particle_index id) const;
+
+        private:
+          /**
+           * Random number generator and an object that describes a
+           * uniform distribution on the interval [0,1]. These
+           * will be used to generate random particle locations.
+           */
+          boost::mt19937            random_number_generator;
+          boost::uniform_01<double> uniform_distribution_01;
```

Comment:
I could, but I would like to have a deterministic behavior, to keep the tests working (currently many tests are failing because I moved this part to the interface but have another RNG in probability_density_function). That is why I initialize them once with a static seed in the constructor. Would that be possible if I create them in a member function?